File: fe/lib/models/course.dart
class Course {
  final String id;
  final String title;
  final String description;
  final int lessons;

  Course({
    required this.id,
    required this.title,
    required this.description,
    required this.lessons,
  });

  factory Course.fromJson(Map<String, dynamic> json) {
    return Course(
      id: json['id'].toString(),
      title: json['title'],
      description: json['description'] ?? '',
      lessons: json['lessons'],
    );
  }
}

File: be/app/Models/Course.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;

class Course extends Model
{
    protected $fillable = [
        'title',
        'description',
        'lessons',
    ];

    public function lessons()
    {
        return $this->hasMany(Lesson::class);
    }

    public function exercises()
    {
        return $this->hasMany(Exercise::class);
    }

    public function listeningItems()
    {
        return $this->hasMany(ListeningItem::class);
    }
}
